tests/GeneratedHydratorTest/Functional/HydratorFunctionalTest.php

```diff
@@ -77,6 +78,37 @@ public function testHydratingNull() : void
         self::assertNull($instance->getProperty0());
     }
 
+    /**
+     * Ensures that the hydrator will not attempt to read unitialized PHP >= 7.4
+     * typed property, which would cause "Uncaught Error: Typed property Foo::$a
+     * must not be accessed before initialization" PHP engine errors.
+     */
+    public function testHydratorWillNotRaisedUnitiliazedTypedPropertyAccessError() : void
+    {
+        if (0 > \version_compare(PHP_VERSION, '7.4.0')) {
+            self::markTestSkipped("This test requires PHP >= 7.4.0");
+        }
+
+        $instance = new ClassWithTypedProperties();
+        $hydrator = $this->generateHydrator($instance);
+
+        $hydrator->hydrate([
+            // 'property0' has a default value, it should keep it.
```

Comment:
Let's add a test in which all fields are overwritten

Reply:
OK